mobile/lib/features/trade/trade_screen.dart

```diff
@@ -282,7 +298,28 @@ class TradeScreen extends StatelessWidget {
 
   Widget createSubmitWidget(PendingOrder pendingOrder, Amount pnl,
       SubmitOrderChangeNotifier submitOrderChangeNotifier, BuildContext context) {
-    Widget body = Column(
+    String bottomText;
+
+    switch (pendingOrder.state) {
+      case PendingOrderState.submittedSuccessfully:
+      case PendingOrderState.submitting:
+        bottomText = "Please wait while the order is being processed. Do not close the app!";
+        break;
+      case PendingOrderState.orderFailed:
+      case PendingOrderState.submissionFailed:
+        bottomText =
+            "Sorry, something went wrong while processing your order. It has been reverted.";
```

Comment:
```suggestion
            "Sorry, we couldn't match your order. Please try again later.";
```

Or something along these lines. Mostly the order fails because the coordinator couldn't find a match.